src/generate/cpp_type.rs

```diff
@@ -15,20 +16,36 @@ use super::{
 // A C# type will be TURNED INTO this
 #[derive(Debug, Clone)]
 pub struct CppType {
+    ty: TypeTag,
     prefix_comments: Vec<String>,
     namespace: String,
     name: String,
-    declarations: Vec<Arc<dyn Writable>>,
-    inherit: Vec<String>,
-    template_line: Option<CppCommentedString>,
-    generic_args: Vec<String>,
-    pub made: bool, // We want to handle metadata generation for stuff too, maybe that goes in the context?
+    declarations: Vec<Arc<CppCommentedString>>,

```

Comment:
Declarations should be writable types, not just CppCommentedStrings. Ex, we could have a series of nested types that are more complex to create here, or generic methods, which are more than just a single CppCommentedString.

Reply:
Yes and no, because I just format the type and insert it in the vector ahead of time. Storing the type itself as a reference requires complications such as lifetimes and borrow checking.